rskj-core/build.gradle

```diff
@@ -1,10 +1,127 @@
 plugins {
     id 'application'
+    id 'checkstyle'
+    id "com.diffplug.spotless" version "6.13.0"
 }
 
 apply plugin: 'maven-publish'
 apply plugin: 'jacoco'
 
+import java.nio.file.Paths
+
+// Define static timestamp
+def staticTimestamp = '2024-09-03T02:17:00'
+ext.timestamp = staticTimestamp
+
+def filteredFiles = getFilteredFiles()
+
+checkstyle {
+    toolVersion = '8.45'
+    configFile = file("$rootDir/config/checkstyle/checkstyle.xml")
+}
+
+spotless {
+    java {
+        googleJavaFormat()
+        indentWithTabs(2)
+        indentWithSpaces(4)
+        target filteredFiles
+
+    }
+}
+
+
+def getFilteredFiles() {
+    def timestamp = ext.timestamp
+
+    def commandLog = [
+            'git', 'log', '--since=' + timestamp, '--name-only', '--pretty=format:', '--abbrev-commit'
+    ]
+    def commandDiff = [
+            'git', 'diff', '--name-only'
+    ]
+    def commandStaged = [
+            'git', 'diff', '--cached', '--name-only'
+    ]
+
+    def outputLog = new ByteArrayOutputStream()
+    def outputDiff = new ByteArrayOutputStream()
+    def outputStaged = new ByteArrayOutputStream()
+
+    exec {
+        commandLine commandLog
+        standardOutput = outputLog
+        errorOutput = outputLog
+        ignoreExitValue = true
+    }
+
+    exec {
+        commandLine commandDiff
+        standardOutput = outputDiff
+        errorOutput = outputDiff
+        ignoreExitValue = true
+    }
+
+    exec {
+        commandLine commandStaged
+        standardOutput = outputStaged
+        errorOutput = outputStaged
+        ignoreExitValue = true
+    }
+
+    def committedFiles = outputLog.toString().trim().split('\n').toList().unique()
+    def uncommittedFiles = outputDiff.toString().trim().split('\n').toList().unique()
+    def stagedFiles = outputStaged.toString().trim().split('\n').toList().unique()
+
+    def allFiles = committedFiles + uncommittedFiles + stagedFiles
+    allFiles = allFiles.unique().findAll { file -> file }
+
+    allFiles = allFiles.findAll { file ->
+        file.endsWith('.java')
+    }
+
+    return allFiles.collect { file -> Paths.get(file).toAbsolutePath().toFile() }
+}
+
+// Method to configure Checkstyle tasks
```

Comment:
nitpick:

I think the name of the method it's pretty much the same what states this comment. I suggest to remove this.

```suggestion
```

Reply:
solved!